### PR TITLE
Integrate Orchard support and latest Rust updates

### LINKED ISSUE #636: Add Orchard Support

### LINKED ISSUE #1367: Add Orchard support

### PR DESCRIPTION
**Pull request by str4d**
Closes #636.
Closes #1367.

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [ ] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [ ] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: Tests/OfflineTests/ZcashRustBackendTests.swift [hunk 1]
```diff
@@ -126,6 +126,11 @@ class ZcashRustBackendTests: XCTestCase {
             XCTFail("not enough transparent receivers")
             return
         }
+
+        // The first address in the wallet is created when the account is created, using
+        // the default receivers specified inside `zcash_client_sqlite`. The remaining
+        // addresses are generated here, using the receivers specified in the Swift SDK's
+        // FFI backend.
```
Comment (str4d) [inline Tests/OfflineTests/ZcashRustBackendTests.swift:133]: This is the reason why only the second and third UAs in the test vectors are updated with an Orchard receiver: the FFI backend now generates Orchard receivers, but `WalletDb::create_account` doesn't (until https://github.com/zcash/librustzcash/pull/1291 is merged).

**Review by LukasKorba — APPROVED** [review]

File: Tests/OfflineTests/ZcashRustBackendTests.swift (hunk 1 above)
Comment (nuttycom) [reply Tests/OfflineTests/ZcashRustBackendTests.swift:133]: Updated to include zcash/librustzcash#1291

**Review by daira — APPROVED** [review]
utACK

**Review by daira — APPROVED** [review]
utACK c98de89613541995eed220677514202aae4904a3

**Pull request merged**